backend/smb/filepool.go

```diff
@@ -0,0 +1,91 @@
+package smb
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/cloudsoda/go-smb2"
+	"golang.org/x/sync/errgroup"
+)
+
```

Comment:
Can you factor w.PoolMu and w.pool out of smbChunkWriter into a new type (`filepool` say) and define it here. It might needs some other stuff like an `f fs.Fs` but I think it is independent of `smbChunkWriter`.

You can make a newFilepool method then this factoring will become a robust re-usable type that is easy to reason about.